src/charm.py

```diff
@@ -135,40 +171,68 @@ def __init__(self, *args):
         # we may reconfigure on ingress changes, so that the path-prefix is updated
         self.framework.observe(self.ingress.on.ready, self._configure_and_start)
         self.framework.observe(self.ingress.on.revoked, self._configure_and_start)
-
+        # we may reconfigure on self-profiling changes, so that scrape config is updated
+        self.framework.observe(
+            self.on["self-profiling-endpoint"].relation_changed, self._configure_and_start
+        )
         # generic status check
         self.framework.observe(self.on.update_status, self._update_status)
 
         # unconditional logic
         self._reconcile()
 
-    @property
-    def _scheme(self) -> str:
-        # TODO: replace with this when integrating TLS
-        # return "https" if self.cert_handler.cert else "http"
-        return "http"
+    ##########################
+    # === PROPERTIES === #
+    ##########################
 
     @property
-    def internal_url(self) -> str:
-        """Return workload's internal URL.
+    def _app_name(self) -> str:
+        """Application name."""
+        return self.app.name
 
-        Used for ingress.
-        """
+    @property
+    def _internal_url(self):
+        """Return workload's internal URL."""
         return f"{self._scheme}://{self._fqdn}:{NGINX_PORT}"
 
     @property
-    def external_url(self) -> str:
+    def _tls_available(self) -> bool:
+        """Return True if tls is enabled and the necessary certs are generated."""
+        if not self.model.relations.get("certificates"):
+            return False
+        cert, key = self.certificates.get_assigned_certificate(
+            certificate_request=self._get_certificate_request_attributes()
+        )
+        return bool(cert and key)
```

Comment:
```suggestion
        return cert and key
```

Reply:
or even:

        return all(self.certificates.get_assigned_certificate(
            certificate_request=self._get_certificate_request_attributes()
        ))

Reply:
also how about calling it `_tls_ready`?